widgetssdk/src/main/java/com/glia/widgets/call/CallActivityIntentHelper.kt

```diff
@@ -6,33 +6,28 @@ import androidx.appcompat.app.AppCompatActivity
 import com.glia.androidsdk.Engagement
 import com.glia.widgets.GliaWidgets
 import com.glia.widgets.core.configuration.EngagementConfiguration
-import java.util.ArrayList
+import com.glia.widgets.helper.getSerializableExtraCompat
 
 internal object CallActivityIntentHelper {
 
-    @JvmStatic
     fun createIntent(context: Context, callConfiguration: CallConfiguration): Intent {
-        val sdkConfiguration = callConfiguration.engagementConfiguration ?: throw NullPointerException("WidgetsSdk Configuration can't be null")
+        val engagementConfiguration =
+            callConfiguration.engagementConfiguration ?: throw NullPointerException("WidgetsSdk Configuration can't be null")
 
         return Intent(context, CallActivity::class.java)
-            .putExtra(GliaWidgets.QUEUE_IDS, sdkConfiguration.queueIds?.let { ArrayList(it) })
-            .putExtra(GliaWidgets.CONTEXT_ASSET_ID, sdkConfiguration.contextAssetId)
-            .putExtra(GliaWidgets.UI_THEME, sdkConfiguration.runTimeTheme)
-            .putExtra(GliaWidgets.SCREEN_SHARING_MODE, sdkConfiguration.screenSharingMode)
+            .putExtra(GliaWidgets.QUEUE_IDS, engagementConfiguration.queueIds?.let(::ArrayList))
+            .putExtra(GliaWidgets.CONTEXT_ASSET_ID, engagementConfiguration.contextAssetId)
+            .putExtra(GliaWidgets.UI_THEME, engagementConfiguration.runTimeTheme)
+            .putExtra(GliaWidgets.SCREEN_SHARING_MODE, engagementConfiguration.screenSharingMode)
             .putExtra(GliaWidgets.MEDIA_TYPE, callConfiguration.mediaType)
             .putExtra(GliaWidgets.IS_UPGRADE_TO_CALL, callConfiguration.isUpgradeToCall)
     }
 
-    @JvmStatic
     fun readConfiguration(activity: AppCompatActivity): CallConfiguration {
         val intent = activity.intent
-        val engagementConfiguration = EngagementConfiguration.Builder().intent(intent).build()
-        val mediaType = intent.getSerializableExtra(GliaWidgets.MEDIA_TYPE) as Engagement.MediaType?
+        val engagementConfiguration = EngagementConfiguration(intent)
+        val mediaType = intent.getSerializableExtraCompat<Engagement.MediaType>(GliaWidgets.MEDIA_TYPE)
         val isUpgradeToCall = intent.getBooleanExtra(GliaWidgets.IS_UPGRADE_TO_CALL, false)
-        return CallConfiguration.Builder()
-            .setEngagementConfiguration(engagementConfiguration)
-            .setMediaType(mediaType)
-            .setIsUpgradeToCall(isUpgradeToCall)
-            .build()
+        return CallConfiguration(engagementConfiguration = engagementConfiguration, mediaType = mediaType, isUpgradeToCall = isUpgradeToCall)
```

Comment:
I think that method property names could be dropped here because they provide 0 benefits to the code readability in this case